chain/network/Cargo.toml

```diff
@@ -50,6 +50,7 @@ time.workspace = true
 delay-detector = { path = "../../tools/delay-detector" }
 near-o11y = { path = "../../core/o11y" }
 near-crypto = { path = "../../core/crypto" }
+near-client-primitives = {path = "../client-primitives"}
```

Comment:
Can we avoid this deps? This adds chain&client stuff to the transitive closure of network deps

Reply:
+1, you can add a new public type to near-network, and then translate it to an isomorphic type in client, like here: https://github.com/near/nearcore/blob/017b433fcf90b1cfc7f3bbf4d66ddb0766763269/chain/client/src/client_actor.rs#L779